src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -1,29 +1,38 @@
+import os
+from proto.import_all_protos import *
+from software.py_constants import MILLISECONDS_PER_SECOND
+import logging
 from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.Qt.QtCore import QTimer
 from pyqtgraph.Qt.QtWidgets import *
 from pyqtgraph import parametertree
 from google.protobuf.json_format import MessageToDict
 from thefuzz import fuzz
 from proto.import_all_protos import *
 from software.thunderscope.common import proto_parameter_tree_util
+from typing import Callable
+from PyQt6.QtWidgets import *
 
 
 class ProtoConfigurationWidget(QWidget):
-
     """Creates a searchable parameter widget that can take any protobuf,
     and convert it into a pyqtgraph ParameterTree. This will allow users
     to modify the values.
 
     """
 
+    # where we are saving the default configuration file
+    DEFAULT_SAVE_DIRECTORY = "/tmp/tbotspython/thunderbots_configurations_proto"
+
     def __init__(
-        self, proto_to_configure, on_change_callback, search_filter_threshold=60,
+        self, on_change_callback, is_yellow, search_filter_threshold=60,
     ):
         """Create a parameter widget given a protobuf
 
         NOTE: This class handles the ParameterRangeOptions
 
         :param proto_to_configure: The protobuf we would like to generate
-                                   a parameter tree for. This should be 
+                                   a parameter tree for. This should be
                                    populated with the default values
```

Comment:
it doesn't seem like you're passing `proto_to_configure` anymore

Reply:
indeed. this is loaded by the file instead of passed by the parameter instead.